drgn/commands/_builtin/crash/_bpf.py

```diff
@@ -0,0 +1,271 @@
+
+        return ",".join(map_ids)
+
+    def __repr__(self) -> str:
+        id_ = self.prog.aux.id.value_()
+        type_ = self.BpfProgType(self.prog.type).name
+        name = self.get_prog_name()
+        try:
+            tail_call_reachable = self.prog.aux.member_("tail_call_reachable").value_()
+        except LookupError:
+            tail_call_reachable = None
+
+        tail_call_desc = " tail_call_reachable" if tail_call_reachable else ""
+
+        return f"{id_:>6}: {type_:32} {name:32}{tail_call_desc}"
+
+
+class BpfTramp(object):
+    def __init__(self, tr: Any) -> None:
+        self.tr = tr
+
+    def get_progs(self) -> Generator[Any, None, None]:
+        if not self.tr:
+            return
+
+        if self.tr.extension_prog:
+            yield self.tr.extension_prog
+            return
+
+        try:
+            for head in self.tr.progs_hlist:
+                for tramp_aux in hlist_for_each_entry(
+                    "struct bpf_prog_aux", head, "tramp_hlist"
+                ):
+                    yield tramp_aux.prog
+        except LookupError:
+            return
+
+
+class BpfMap(object):
+    def __init__(self, bpf_map: Any, BpfProgType: Any, BpfMapType: Any) -> None:
+        self.map = bpf_map
+        self.BpfMapType = BpfMapType
+        self.BpfProgType = BpfProgType
+
+    def inspect_owner(self, owner: Any) -> str:
+        type_ = self.BpfProgType(owner.type).name
+        jited = " JITed" if owner.jited.value_() else ""
+        return f"{type_:32}{jited}"
+
+    def get_owner(self) -> str:
+        try:
+            owner = self.map.member_("owner")
+            return self.inspect_owner(owner)
+        except LookupError:
+            return ""
+
+    def get_map_flags_hex(self) -> str:
+        try:
+            # Map flags are u32, print as 8-digit hex
+            return f"{self.map.map_flags.value_():08x}"
+        except LookupError:
+            return "00000000"
+
+    def __repr__(self) -> str:
+        id_ = self.map.id.value_()
+        type_ = self.BpfMapType(self.map.map_type).name
+        name = self.map.name.string_().decode()
+
+        return f"{id_:>6}: {type_:32} {name:32}"
+
+
+@crash_command(
+    description="display all eBPF programs and maps",
+)
+def _crash_cmd_bpf(
+    prog: Program, name: str, args: argparse.Namespace, **kwargs: Any
+) -> None:
+    BpfMapType = enum_type_to_class(prog.type("enum bpf_map_type"), "BpfMapType")
+    BpfProgType = enum_type_to_class(prog.type("enum bpf_prog_type"), "BpfProgType")
+    BpfAttachType = enum_type_to_class(
+        prog.type("enum bpf_attach_type"), "BpfAttachType"
+    )
+    BpfLinkType = enum_type_to_class(prog.type("enum bpf_link_type"), "BpfLinkType")
+    BpfProgTrampType = enum_type_to_class(
+        prog.type("enum bpf_tramp_prog_type"), "BpfProgTrampType"
+    )
+
+    print(
+        f"{'ID':>4}  {'BPF_PROG':<18} {'BPF_PROG_AUX':<18} {'BPF_PROG_TYPE':<18} "
+        f"{'TAG':<18} {'USED_MAPS'}"
+    )
```

Comment:
I'd prefer use of `drgn.helpers.common.format.print_table()` rather than manual formatting. There are lots of example in `drgn/commands`.